tests/storage-states/get-secrets.setup.ts

```diff
@@ -0,0 +1,11 @@
+require("dotenv").config();
+import { test as setup } from "../fixtures/fixtures";
+import { getSecrets } from "../authentication/retrieve-aws-creds-ci";
+
+setup("get secrets", async () => {
+    const { username, password, moviedb_access_token } = await getSecrets();
+
+    process.env.MOVIEDB_USERNAME = process.env.MOVIEDB_USERNAME || username;
+    process.env.MOVIEDB_PASSWORD = process.env.MOVIEDB_PASSWORD || password;
+    process.env.MOVIEDB_ACCESS_TOKEN = process.env.MOVIEDB_ACCESS_TOKEN || moviedb_access_token;
+});
```

Comment:
For steps like these, that you only want to execute once/test run, you could use the globalSetup part of the Playwright Config.
([Related Documentation](https://playwright.dev/docs/test-global-setup-teardown#option-2-configure-globalsetup-and-globalteardown))

This way you can get rid of the test-like implementation and simply call the necessary function(s), do the necessary steps to achieve the same state.

Reply:
I was not aware of it, but it's definitely a better solution, thank you :)

Reply:
If all the environment variables are set, there is no need to get the secrets from AWS.